src/FakeItEasy/Creation/CastleDynamicProxy/CastleDynamicProxyGenerator.cs

```diff
@@ -19,13 +21,15 @@ internal class CastleDynamicProxyGenerator
         private static readonly ProxyGenerator ProxyGenerator = new ProxyGenerator();
         private readonly CastleDynamicProxyInterceptionValidator interceptionValidator;
 
+#if FEATURE_SECURITY_PERMISSIONS
         [SuppressMessage("Microsoft.Performance", "CA1810:InitializeReferenceTypeStaticFieldsInline", Justification = "No field initialization.")]
         static CastleDynamicProxyGenerator()
         {
 #pragma warning disable 618
             AttributesToAvoidReplicating.Add(typeof(SecurityPermissionAttribute));
 #pragma warning restore 618
         }
+#endif

```

Comment:
@blairconrad there is no need to manually add `SecurityPermissionAttribute` anymore [since Castle Core 2.5.2](https://github.com/castleproject/Core/commit/cf88a405a58f1c4f0b27a0d150f688fb25b11498) because it is already [added by default by DynamicProxy](https://github.com/castleproject/Core/blob/abce66323afb2944a3af7c5a0215739ebf9b22ef/src/Castle.Core/DynamicProxy/Generators/AttributesToAvoidReplicating.cs#L32). DP should handle any special case attributes that break the runtime.